### ISSUE TITLE
How to disable git in theme? It is very slow on a big repo. Even slower than git status.

### ISSUE THREAD

**Issue by ivakyb** [oh-my-fish/oh-my-fish#686]

**Comment by bobthecow** [comment]
It depends on the theme. You should open an issue on the repo of the theme you're having trouble with.

**Comment by ivakyb** [comment]
Ok. Thanks. But I have this problem with almost all themes because all of them calls `git status --porcelain` or something similar.

**Comment by bobthecow** [comment]
right. but the way to disable git in themes depends on the theme. for example, bobthefish has three or four ways to make git less demanding, or to disable it entirely on a per-repo or system-wide basis, but none of those apply to any theme besides bobthefish.

**Comment by scorphus** [comment]
I'm reopening this issue to get it solved on HackIllinois 2019, hopefully!

**Comment by kfkonrad** [comment]
I think adding variables to en-/disable certain parts of a prompt may be a good idea, at least where it really impacts performance. The code you've written for the `scorphish` theme seems like a good, simple solution.

Can we make the use of such variables a standard among themes, at least for the ones in the oh-my-fish org? We can then document this new de facto standard, spread the word and hopefully get others to adopt it.

**Comment by bobthecow** [comment]
There is a de-facto standard for such variable names in bobthefish. Several other themes have adopted the same names for disabling equivalent features. See https://github.com/oh-my-fish/theme-bobthefish#configuration

**Comment by scorphus** [comment]
> Can we make the use of such variables a standard among themes, at least for the ones in the oh-my-fish org? We can then document this new de facto standard, spread the word and hopefully get others to adopt it.

Yes, I do think we can. That may be one way to solve this. As mentioned just above, `scorphish` now follows the same variable names as in `bobthefish`. That could be done to other themes, provided we keep backwards compatibility (support to "old-fashion" variable names).
